peps/pep-0744.rst

```diff
@@ -0,0 +1,566 @@
+with more domain expertise in collaboration with those maintaining the JIT, who
+have little experience with the inner workings of these tools.
+
+Security Implications
+=====================
+
+This JIT, like any JIT, produces large amounts of executable data at runtime.
+This introduces a potential new attack surface to CPython, since a malicious
+actor capable of influencing the contents of this data is therefore capable of
+executing arbitrary code. This is a `well-known vulnerability
+<https://en.wikipedia.org/wiki/Just-in-time_compilation#Security>`_ of JIT
+compilers.
+
+In order to mitigate this risk, the JIT has been written with best practices in
+mind. In particular, the data in question is not exposed by the JIT compiler to
+other parts of the program while it remains writeable, and at *no* point is the
+data both |wx|_.
+
+.. Apparently this how you hack together a formatted link:
+
+.. |wx| replace:: writable *and* executable
+.. _wx: https://en.wikipedia.org/wiki/W%5EX
+
+The nature of template-based JITs also seriously limits the kinds of code that
+can be generated, further reducing the likelihood of a successful exploit. As an
+additional precaution, the templates themselves are stored in static, read-only
+memory.
+
+However, it would be naive to assume that no possible vulnerabilities exist in
+the JIT, especially at this early stage. The author is not a security expert,
+but is available to join or work closely with the Python Security Response Team
+to triage and fix security issues as they arise.
+
+Apple Silicon
+--------------
+
+Though difficult to test without actually signing and packaging a macOS release,
+it *appears* that macOS releases should `enable the JIT Entitlement for the
+Hardened Runtime
+<https://developer.apple.com/documentation/apple-silicon/porting-just-in-time-compilers-to-apple-silicon#Enable-the-JIT-Entitlement-for-the-Hardened-Runtime>`_.
+
+This shouldn't make *installing* Python any harder, but may add additional steps
+for release managers to perform.
+
+How to Teach This
+=================
+
+Choose the sections that best describe you:
+
+- **If you are a Python programmer or end user...**
+  
+  - ...nothing changes for you. Nobody should be distributing JIT-enabled
+    CPython interpreters to you while it is still an experimental feature. Once
+    it is non-experimental, you will probably notice slightly better performance
+    and slightly higher memory usage. You shouldn't be able to observe any other
+    changes.
+
+- **If you maintain third-party packages...**
+
+  - ...nothing changes for you. There are no API or ABI changes, and the JIT is
+    not exposed to third-party code. You shouldn't need to change your CI
+    matrix, and you shouldn't be able to observe differences in the way your
+    packages work when the JIT is enabled.
+
+- **If you profile or debug Python code...**
+
+  - ...nothing changes for you. All Python profiling and tracing functionality
+    remains.
```

Comment:
This seems to contradict the earlier statement of out of process profilers and debuggers?

Reply:
The distinction here is Python debuggers using `sys.monitoring`/`sys.settrace`/`sys.setprofile` and native tools that trace/profile C code, like `perf` or `gdb`.

Maybe the difference should be rephrased or made clearer. Any suggestions?

Reply:
In other words. If you only look at Python frames, you're fine. If you look at C frames at all, stuff may break.

Reply:
Okay, so sounds like things like `py-spy` (that do ptrace -> find interpreter state -> find frame objects) will continue to work, but `py-spy --native` (that do ptrace -> libunwind -> resolve symbols -> merge with python stacktrace) may break, and things that rely on the py312 perf trampoline may break?

If that's right, I think the wording here is fine, but maybe in the "debugging" section we could more explicitly define "Tools that profile and debug Python" to include out-of-process py-spy like tools?